_posts/2024-12-16-edge-computing-tech.md

```diff
@@ -0,0 +1,59 @@
+---
+layout: post
+title: Edge computing, some ways to make it happen
+author: msasurinen
+excerpt: >
+  In the last blog I introduced the concept of edge computing and outlined its benefits. In this blog post I want to go over the some of the enabling technologies, focusing on how your trusted cloud providers could help us in this matter. 
+tags:
+ - Edge computing
+ - Iot
+ - Intelligent Devices 
+---
+# Getting things connected 
+ 
+Getting the devices connected to our cloud environment is usually the first step in integrating the edge into our system, or atleast its often a neccessary step. For this function the go to Azure resource is **Azure IoT Hub**. 
+IoT Hub is a managed service that handles messaging and device management. Incoming messages can be enriched with metadata and relayed to other Azure services quite easily. 
+It also gives your devices an identity in your cloud, it includes a device identity registry where you create identities for your IoT devices. Authentication is done via SAS tokens or X.509 certificates, with the latter being more secure but requiring the device to securely store the private key. 
+Once provisioned, devices can send and receive messages. In this context cloud-to-device (C2D) messages are called direct methods, which are synchronous calls to methods defined on the device, say setting a temperature threshold.
+IoT Hub also provides a device twin, which is a JSON document with metadata (like last connection timestamp), desired properties, and reported properties. 
+This acts as another way to set device parameters like the aforementioned temperature thresholds. The device will receive updates when the twin changes, and it can report back values that are then accessible from the Iot hub. For a code example, check out the [IoT Hub documentation.](https://learn.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-device-twins)  
+
+These same features can be found from **AWS Iot Core** as Device Shadows and IoT jobs
+
+IoT Hub and IoT Core offer strong connectivity features, but they don’t provide much control over the workloads running on devices. That’s where services like **Azure IoT Edge** comes in. 
+It is an extension of sorts of IoT Hub. Iot Edge containing two docker containers that you set up locally. 
```

Comment:
Consistency on IoT vs Iot and AWS and Aws throughout the text would be ideal 👍🏻

Reply:
Yeah thanks for pointing that out, have gotten blind to it while writing